api/RestfulService.php

```diff
@@ -83,7 +101,7 @@ public static function set_default_proxy($proxy, $port = 80, $user = "", $passwo
 			'Use the "RestfulService.default_curl_options" config setting instead, '
 				. 'with direct reference to the CURL_* options'
 		);
-		config::inst()->update('RestfulService', 'default_proxy', array(
+		config::inst()->update(get_called_class(), 'default_proxy', array(
```

Comment:
This should really be static::config() too - just noticed that config is lowercase.